GeorgiaTech/Server/Controllers/VolumeController.cs

```diff
@@ -0,0 +1,194 @@
+                return changedRows;
+
+            }
+            catch
+            {
+                transaction.Rollback();
+                throw;
+
+            }
+        }
+
+        /// <summary>
+        /// Find all the volumes in the database
+        /// </summary>
+        /// <returns>A list of volumes</returns>
+        public List<Volume> FindAll()
+        {
+            IQueryable<Volume> volumes = _context.Volumes
+                .Include(v => v.Material)
+                    .ThenInclude(m => m.MaterialAuthors)
+                        .ThenInclude(ma => ma.Author)
+                .Include(v => v.Material)
+                    .ThenInclude(m => m.MaterialSubjects)
+                        .ThenInclude(ms => ms.MaterialSubject)
+                .Include(v => v.HomeLocation)
+                    .ThenInclude(a => a.Zip)
+                .Include(v => v.CurrentLocation)
+                    .ThenInclude(a => a.Zip);
+
+            return volumes.ToList();
+        }
+
+        /// <summary>
+        /// Find all volumes for a specific material
+        /// </summary>
+        /// <param name="materialId"></param>
+        /// <returns>A list of volumes for the specified material</returns>
+        public List<Volume> FindVolumesForMaterial(int materialId)
+        {
+            try
+            {
+                IQueryable<Volume> volumes = _context.Volumes.Where(v => v.MaterialId == materialId)
+              .Include(v => v.Material)
+                  .ThenInclude(m => m.MaterialAuthors)
+                      .ThenInclude(ma => ma.Author)
+              .Include(v => v.Material)
+                  .ThenInclude(m => m.MaterialSubjects)
+                      .ThenInclude(ms => ms.MaterialSubject)
+              .Include(v => v.HomeLocation)
+                  .ThenInclude(a => a.Zip)
+              .Include(v => v.CurrentLocation)
+                  .ThenInclude(a => a.Zip);
+
+                return volumes.ToList();
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+          
+        }
+
+        public Volume FindByID(int ID)
+        {
+            try
+            {
+                var volume = _context.Volumes
+        .Include(v => v.Material)
+            .ThenInclude(m => m.MaterialAuthors)
+                .ThenInclude(ma => ma.Author)
+        .Include(v => v.Material)
+            .ThenInclude(m => m.MaterialSubjects)
+            .ThenInclude(ms => ms.MaterialSubject)
+        .Include(v => v.HomeLocation)
+            .ThenInclude(a => a.Zip)
+         .Include(v => v.CurrentLocation)
+             .ThenInclude(a => a.Zip)
+         .Single(v => v.VolumeId == ID);
+
+                return volume;
+            }
+            catch
+            {
+                throw new NullReferenceException(message: $"Volume with id: {ID} not found");
+            }
+    
+        }
+
+        public Volume FindByType(Volume t)
```

Comment:
if this is not going to be implemented ( as others are actually useful) - add the [obsolete] above it.